### PR TITLE
add `onCancel` delegation to Session

### PR DESCRIPTION
**Pull request by slavikbialik**

### CODE REVIEW

File: src/api/cancel.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * A request to end a {@link Session} (incoming CANCEL).
+ * @public
+ */
+export class Cancel {
+  /** @internal */
+  public constructor(private incomingCancelRequest: IncomingRequestMessage) {}
+
+  /** Incoming ACK request message. */
```
Comment (alex-dizengof) [inline src/api/cancel.ts:11]: /** Incoming ACK request message. */
Why ACK?

File: src/api/invitation.ts [hunk 2]
```diff
@@ -420,6 +421,14 @@ export class Invitation extends Session {
   public _onCancel(message: IncomingRequestMessage): void {
     this.logger.log("Invitation._onCancel");
 
+    // validate state for CANCEL delegation
+    if (this.state === SessionState.Initial || this.state === SessionState.Establishing) {
```
Comment (alex-dizengof) [inline src/api/invitation.ts:425]: Why do we need to check the State not just send the Cancel?
What problem it solves?

File: src/api/cancel.ts (hunk 1 above)
Comment (slavikbialik) [reply src/api/cancel.ts:11]: Yep, it's a copy-paste from another object, I will fix.

File: src/api/invitation.ts (hunk 2 above)
Comment (slavikbialik) [reply src/api/invitation.ts:425]: After talking with Eric from SIP.js he said he'll only accept this addition only if I won't try to bypass the RFC.
As CANCEL should not arrive in ESTABLISHED state, and if it does according the RFC it should ignore it and not process it at all. 
That's why the delegation part will only work in INITIAL and ESTABLISHING.

Initial - is when the CT is still ringing.
Establishing - is when the CT SIP.js is still structing the 200OK but hasn't sent it out yet.
Established - is when the 200OK is already sent.

This step is not for processing the CANCEL, it is being done already... this step is for processing the delegation part only.

**Review by shmuel7 — APPROVED** [review]

**Review by romanmalih — APPROVED** [review]

**Pull request merged**